Sources/Core/ApiClient/ApiClient.swift

```diff
@@ -25,6 +25,11 @@ struct ApiClient {
     as type: Value.Type
   ) async throws -> (value: Value, response: URLResponse) {
     let (data, response) = try await self.request(route)
-    return (try self.decoder.decode(type, from: data), response)
+    let decodedResponse = try await decode(data, as: type)
+    return (decodedResponse, response)
```

Comment:
```suggestion
    let value = try await decode(data, as: type)
    return (value, response)
```
Maybe value is a more suitable name here. Thoughts?

Reply:
Good call, value is a better name